docker_cli/_docker_mamba.py

```diff
@@ -0,0 +1,235 @@
+
+
+def mamba_lockfile_command(
+    env_name: str,
+) -> str:
+    """
+    Returns a command to generate a lockfile with micromamba.
+
+    Parameters
+    ----------
+    env_name : str
+        The name of the environment for which a Dockerfile should be generated.
+
+    Returns
+    -------
+    str
+        The command.
+    """
+    cmd = f"micromamba env export --name {env_name} --explicit --no-md5"
+    return cmd
+
+
+def micromamba_docker_lines():
+    return textwrap.dedent(
+        """
+        ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+        USER $MAMBA_USER
+        """
+    ).strip()
+
+
+@overload
+def _mamba_spec_command(
+    command: str,
+    *,
+    channels: Optional[Iterable[str]],
+    packages: Iterable[str],
+    env_name: Optional[str] = ...,
+) -> str:
+    ...
+
+
+@overload
+def _mamba_spec_command(
+    command: str,
+    *,
+    channels: Optional[Iterable[str]],
+    specfile: os.PathLike[str] | str,
+    env_name: Optional[str] = ...,
+) -> str:
+    ...
+
+
+def _mamba_spec_command(
+    command, *, channels, specfile=None, packages=None, env_name=None
+) -> str:
+    if (specfile is not None) and (packages is not None):
+        raise ValueError("Specfile and packages both given. Use only one.")
+    if command not in ["install", "create"]:
+        raise ValueError(
+            "Improper value given for micromamba specs installation. "
+            'Please use one of "install" or "create".'
+        )
+
+    # If a name was given for the environment, an argument will be added to all
+    # micromamba installation commands.
+    if env_name is None:
+        name_arg: str = ""
+    else:
+        name_arg = f" -n {env_name}"
+
+    # Put together the channels argument
+    if channels is not None:
+        channels_arg = f" -c {shlex.join(channels)} --override-channels"
+
+    # If a specfile is given, give the instructions for copying and installing a
+    # specfile.
+    if specfile is not None:
+        install_command = textwrap.dedent(
+            f"""
+            COPY {specfile} /tmp/spec-file.txt
+            RUN micromamba {command}{name_arg}{channels_arg} -y -f /tmp/spec-file.txt \\
+             && rm /tmp/spec-file.txt \\
+        """
+        ).strip()
+    # Otherwise packages were given, so give the instructions for installing the
+    # packages.
+    else:
+        packages_str = shlex.join(packages)
+        install_command = textwrap.dedent(
+            f"""
+            RUN micromamba {command}{name_arg}{channels_arg} -y {packages_str} \\
+             && micromamba update --all \\
+        """
+        ).strip()
+
+    # Assemble this command into a portion of a Dockerfile string.
+    cmd: str = (
+        "ARG MAMBA_DOCKERFILE_ACTIVATE=1\n\n"
+        + install_command
+        + "\n"
+        + " && micromamba clean --all --yes"
+    )
+
+    return cmd
+
+
+def _mamba_install_prefix():
+    return "FROM mambaorg/micromamba:1.3.1 as micromamba"
```

Comment:
By convention, Dockerfile commands should be all-caps

```suggestion
    return "FROM mambaorg/micromamba:1.3.1 AS micromamba"
```